src/main/java/com/cbfacademy/apiassessment/DestinationAddressService.java

```diff
@@ -0,0 +1,87 @@
+package com.cbfacademy.apiassessment;
+
+import org.springframework.stereotype.Service;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@Service
+public class DestinationAddressService {
+
+    private static final String DESTINATION_FILE_PATH = "src/main/resources/destination.json";
+    private final ObjectMapper objectMapper;
+
+    public DestinationAddressService() {
+        this.objectMapper = new ObjectMapper().enable(SerializationFeature.INDENT_OUTPUT);
+    }
+
+    public List<DestinationAddress> getDestinationAddresses() {
+        try {
+            DestinationAddress[] addressesArray = objectMapper.readValue(new File(DESTINATION_FILE_PATH),
+                    DestinationAddress[].class);
+            return new ArrayList<>(Arrays.asList(addressesArray));
+        } catch (IOException e) {
+            e.printStackTrace();
+            return new ArrayList<>();
+        }
+    }
+
+    public DestinationAddress getDestinationAddress(int destinationId) {
+        List<DestinationAddress> addresses = getDestinationAddresses();
+        for (DestinationAddress address : addresses) {
+            if (address.getId() == destinationId) {
+                return address;
+            }
+        }
+        return null; // Address with the given ID not found
+    }
+
+    public DestinationAddress addDestinationAddress(DestinationAddress address) {
+        List<DestinationAddress> addresses = getDestinationAddresses();
+        int newId = addresses.stream().mapToInt(DestinationAddress::getId).max().orElse(0) + 1;
+        address.setId(newId);
```

Comment:
This is where you should create a new `DestinationAddress` instance and pass `newId` to the constructor, along with all the properties from `address`